src/daria/image/image.py

```diff
@@ -193,26 +189,33 @@ def write(
             name (str): name of image
             path (str): tpath to image (only folders)
             file_format (str): file ending, deifning the image format
-            save_metadata (bool): controlling whether metadata is stored
-            metadata_path (str): path to metadata (only folders); the metadata file
-                has the same name as the image (and a .txt ending)
         """
         # cv2 requires BGR format
         self.toBGR()
-        assert self.colorspace == "bgr"
+        assert self.colorspace == "BGR"
 
         # Write image, using the conventional matrix indexing
         cv2.imwrite(str(Path(path + name + file_format)), self.img)
         print("Image saved as: " + str(Path(path + name + file_format)))
 
-        # Write meta data
-        if save_metadata:
-            f = open(str(Path(metadata_path + name + ".txt")), "w")
-            f.write("Origo: " + str(self.origo) + "\n")
-            f.write("Width: " + str(self.width) + "\n")
-            f.write("Height: " + str(self.height) + "\n")
-            f.write("Dimension: " + str(self.dim) + "\n")
-            f.close()
+    def update_metadata(self) -> None:
+        self.metadata: dict = {
+            "width": self.width,
+            "height": self.height,
+            "origo": self.origo,
+            "color_space": self.colorspace,
+        }
+
+    def write_metadata_to_file(self, path: str) -> None:
+        """
+        Writes the metadata dictionary to a json-file.
+
+        Arguments:
+            path (str): path to the json file
+        """
+
+        with open(str(Path(path)), "w") as outfile:
+            json.dump(self.metadata, outfile, indent=4)
```

Comment:
Oh, now I see that you wanted to keep the organization of the metadata within `Image`... Let's keep it here for now. There is no other more natural place for it right now.

Reply:
Haha, yes. But I agree. it is debatable how we would like to do all of this business.